inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/main/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvFormatDeserializer.java

```diff
@@ -0,0 +1,334 @@
+            @Nonnull Character kvDelimiter,
+            @Nullable Character escapeChar,
+            @Nullable Character quoteChar,
+            @Nullable String nullLiteral,
+            @Nonnull Boolean ignoreErrors) {
+        this(
+                rowFormatInfo,
+                timeFieldName,
+                attributesFieldName,
+                charset,
+                delimiter,
+                entryDelimiter,
+                kvDelimiter,
+                escapeChar,
+                quoteChar,
+                nullLiteral,
+                InLongMsgUtils.getDefaultExceptionHandler(ignoreErrors));
+    }
+
+    public InLongMsgTlogKvFormatDeserializer(
+            @Nonnull RowFormatInfo rowFormatInfo,
+            @Nullable String timeFieldName,
+            @Nullable String attributesFieldName,
+            @Nonnull String charset,
+            @Nonnull Character delimiter,
+            @Nonnull Character entryDelimiter,
+            @Nonnull Character kvDelimiter,
+            @Nullable Character escapeChar,
+            @Nullable Character quoteChar,
+            @Nullable String nullLiteral,
+            @Nonnull FailureHandler failureHandler) {
+        super(failureHandler);
+
+        this.rowFormatInfo = rowFormatInfo;
+        this.timeFieldName = timeFieldName;
+        this.attributesFieldName = attributesFieldName;
+        this.charset = charset;
+        this.delimiter = delimiter;
+        this.entryDelimiter = entryDelimiter;
+        this.kvDelimiter = kvDelimiter;
+        this.escapeChar = escapeChar;
+        this.quoteChar = quoteChar;
+        this.nullLiteral = nullLiteral;
+
+        this.converters = Arrays.stream(rowFormatInfo.getFieldFormatInfos())
+                .map(formatInfo -> FieldToRowDataConverters.createConverter(
+                        TableFormatUtils.deriveLogicalType(formatInfo)))
+                .toArray(FieldToRowDataConverters.FieldToRowDataConverter[]::new);
+    }
+
+    @Override
+    public TypeInformation<RowData> getProducedType() {
+        return InLongMsgUtils.decorateRowDataTypeWithNeededHeadFields(timeFieldName,
+                attributesFieldName, (RowType) TableFormatUtils.deriveLogicalType(rowFormatInfo));
+    }
+
+    @Override
+    protected InLongMsgHead parseHead(String attr) throws Exception {
+        return InLongMsgTlogKvUtils.parseHead(attr);
+    }
+
+    @Override
+    protected List<InLongMsgBody> parseBodyList(byte[] bytes) throws Exception {
+        return Collections.singletonList(
+                InLongMsgTlogKvUtils.parseBody(
+                        bytes,
+                        charset,
+                        delimiter,
+                        entryDelimiter,
+                        kvDelimiter,
+                        escapeChar,
+                        quoteChar));
+    }
+
+    @Override
+    protected List<RowData> convertRowDataList(InLongMsgHead head, InLongMsgBody body) throws Exception {
+        GenericRowData rowD =
```

Comment:
rowD is vague, maybe genericData ? or genericRow

Reply:
done change to genericRowData